SS2-Project/Assets/Starstorm2/Modules/Pickups/Equipments/AffixPurple.cs

```diff
@@ -1,22 +1,171 @@
+        //{
+        //    return false;
+        //}
+
+        public override List<EliteDef> EliteDefs => _eliteDefs;
+        private List<EliteDef> _eliteDefs;
+
+        public override NullableRef<List<GameObject>> ItemDisplayPrefabs => null;
+
+        public override EquipmentDef EquipmentDef => _equipmentDef;
+        private EquipmentDef _equipmentDef;
+
+        public static DotController.DotIndex index;
+
+        public BuffDef _buffPurplePoisonBuildup; // TODO: { get; } = SS2Assets.LoadAsset<BuffDef>("bdPoisonBuildup", SS2Bundle.Indev);
+        public BuffDef _buffPurplePoisonDebuff; // { get; } = SS2Assets.LoadAsset<BuffDef>("bdPurplePoison", SS2Bundle.Indev);
+        public BuffDef _buffAffixPurple; //{ get; } = SS2Assets.LoadAsset<BuffDef>("bdElitePurple", SS2Bundle.Indev);
+
+        public override void Initialize()
+        {
+            // TODO: Do I need a separate dot for poisondebuff buffdef? idk
+            index = DotAPI.RegisterDotDef(0.25f, 0.18f, DamageColorIndex.DeathMark, _buffPurplePoisonBuildup);
+        }
 
-        public override List<MSEliteDef> EliteDefs { get; } = new List<MSEliteDef>
+        public override bool IsAvailable(ContentPack contentPack)
         {
-            SS2Assets.LoadAsset<MSEliteDef>("edPurple", SS2Bundle.Indev),
-            SS2Assets.LoadAsset<MSEliteDef>("edPurpleHonor", SS2Bundle.Indev)
-        };
+            return false;
+        }
+
+        public override IEnumerator LoadContentAsync()
+        {
+            yield break;
+        }
 
-        public override bool FireAction(EquipmentSlot slot)
+        public override bool Execute(EquipmentSlot slot)
         {
             return false;
         }
-    }*/
+
+        public override void OnEquipmentLost(CharacterBody body)
+        {
+        }
+
+        public override void OnEquipmentObtained(CharacterBody body)
+        {
+        }
+
+        public class PurplePoisonBuildup : BaseBuffBehaviour
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdPoisonBuildup;
+        }
+
+        public class PurplePoisonDebuff : BaseBuffBehaviour
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdPurplePoison;
+        }
+
+        public sealed class AffixPurpleBehavior : BaseBuffBehaviour, IOnDamageDealtServerReceiver
+        {
+            [BuffDefAssociation]
+            private static BuffDef GetBuffDef() => SS2Content.Buffs.bdElitePurple;
+
+            private float prevInterval = 0.0f;
+
+            private static float projectileLaunchInterval = 30f;
+
+            // Fields for launching projectiles
+            private int radius = 4;
+            private static int numProjectiles = 2;
+            private float damage;
+            private static int duration = 5;
+
+            public void OnDamageDealtServer(DamageReport damageReport)
```

Comment:
Check if the behaviour is enabled by returning early after checking if "HasAnyStacks" is true.

Excerpt from the documentation on buff behaviour:

``
Due to the nature of Buffs being Ephemeral, this Behaviour is not destroyed when the buff is removed, but instead it becomes disabled. Due to this reason, most, if not all behaviour interfaces will still get called even if the behaviour itself is disabled. For this reason it is recommended to break from the function early by checking if <see cref="HasAnyStacks"/> is false``